apollo-runtime/src/main/java/com/apollographql/android/impl/BufferedResponseReader.java

```diff
@@ -64,15 +67,28 @@
         value = readCustomType((Field.CustomTypeField) field);
         break;
       case CONDITIONAL:
-        value = readConditional((Field.ConditionalTypeField) field);
+        value = readConditional((Field.ConditionalTypeField) field, operation.variables());
```

Comment:
The parsing for conditional is slightly strange 

When we call readConditional, the field is actually `__typename` and not the conditional field. However, we will need to store the `_typename` for reading from the cache. Thus, read conditional actually stores two values at once, and can't follow as clean as a pattern as every other type of field.